lib/pacemaker/pcmk_sched_utils.c

```diff
@@ -787,3 +787,59 @@ pcmk__new_logger(void)
     pcmk__register_lib_messages(out);
     return out;
 }
+
+/*!
+ * \internal
+ * \brief Check whether a resource has reached its migration threshold on a node
+ *
+ * \param[in] rsc       Resource to check
+ * \param[in] node      Node to check
+ * \param[in] data_set  Cluster working set
+ *
+ * \return Failed resource (possibly a parent of \p rsc) if migration threshold
+ *         has been reached, or NULL otherwise
+ */
+pe_resource_t *
+pcmk__threshold_reached(pe_resource_t *rsc, pe_node_t *node,
+                        pe_working_set_t *data_set)
+{
```

Comment:
The name seems weird to me.  I'd expect pcmk__threshold_reached to return a bool like the old function did.  Maybe something like pcmk__rsc_over_threshold?  That kind of seems bool-like to me too.

Reply:
I could make it return bool, and take the failed resource as an output argument

Reply:
Added commit on top for this

Reply:
That works for me.